### ISSUE TITLE
dilated tf.keras.layers.Conv2D can't estimate output shape

### ISSUE THREAD

**Issue by breadbread1984** [tensorflow/tensorflow#31109]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.0.0-beta1
- Python version: 3.6.8
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**

when creating tf.keras.Model with functional api, dilated convolution of tf.keras.layers can't estimate the output shape.

**Describe the expected behavior**

the dilated convolution can estimate the output shape as convolution op does.

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.

```python
import tensorflow as tf

input= tf.keras.Input([100,100,3])
results = tf.keras.layers.Conv2D(filters = 10, kernel_size = (3,3), padding = 'same', dilation_rate = 2)(input)
print(results.shape)
```

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by ravikyram** [comment]
I have tried on colab with TF version 2.0 beta1 and was able to reproduce the issue.Please, find the [gist](https://colab.research.google.com/drive/10utd0A3GuBGAvIf8RnMQVFRp2QjAbFPC) here.Thanks!

**Comment by ymodak** [comment]
This is fixed in latest 2.0 nightly build version' 2.0.0-dev20190729'. Thanks!

**Comment by breadbread1984** [comment]
the same problem still exists in dilated Conv1D.
